lib/ansible/modules/cloud/google/gcp_dns_resource_record_set.py

```diff
@@ -0,0 +1,466 @@
+                                   'dns#resourceRecordSetsListResponse',
+                                   'rrsets')
+    changed = False
+
+    if fetch:
+        if state == 'present':
+            if is_different(module, fetch):
+                fetch = update(module, self_link(module), kind, fetch)
+        else:
+            delete(module, self_link(module), kind, fetch)
+            fetch = {}
+            changed = True
+    else:
+        if state == 'present':
+            fetch = create(module, collection(module), kind)
+            changed = True
+
+    if fetch:
+        fetch.update({'changed': changed})
+    else:
+        fetch = {'changed': changed}
+
+    module.exit_json(**fetch)
+
+
+def create(module, link, kind):
+    change = create_change(None, updated_record(module), module)
+    change_id = int(change['id'])
+    if change['status'] == 'pending':
+        wait_for_change_to_complete(change_id, module)
+    return fetch_wrapped_resource(module, 'dns#resourceRecordSet',
+                                  'dns#resourceRecordSetsListResponse',
+                                  'rrsets')
+
+
+def update(module, link, kind, fetch):
+    change = create_change(fetch, updated_record(module), module)
+    change_id = int(change['id'])
+    if change['status'] == 'pending':
+        wait_for_change_to_complete(change_id, module)
+    return fetch_wrapped_resource(module, 'dns#resourceRecordSet',
+                                  'dns#resourceRecordSetsListResponse',
+                                  'rrsets')
+
+
+def delete(module, link, kind, fetch):
+    change = create_change(fetch, None, module)
+    change_id = int(change['id'])
+    if change['status'] == 'pending':
+        wait_for_change_to_complete(change_id, module)
+    return fetch_wrapped_resource(module, 'dns#resourceRecordSet',
+                                  'dns#resourceRecordSetsListResponse',
+                                  'rrsets')
+
+
+def resource_to_request(module):
+    request = {
+        u'kind': 'dns#resourceRecordSet',
+        u'managed_zone': module.params['managed_zone'],
+        u'name': module.params['name'],
+        u'type': module.params['type'],
+        u'ttl': module.params['ttl'],
+        u'rrdatas': module.params['target'],
+    }
+    return_vals = {}
+    for k, v in request.items():
+        if v:
+            return_vals[k] = v
+
+    return return_vals
+
+
+def fetch_resource(module, link, kind):
+    auth = GcpSession(module, 'g')
+    return return_if_object(module, auth.get(link), kind)
+
+
+def fetch_wrapped_resource(module, kind, wrap_kind, wrap_path):
+    result = fetch_resource(module, self_link(module), wrap_kind)
+    if result is None or wrap_path not in result:
+        return None
+
+    result = unwrap_resource(result[wrap_path], module)
+
+    if result is None:
+        return None
+
+    if result['kind'] != kind:
+        raise "Incorrect result: {0} (expected {1})".format(result['kind'], kind)
```

Comment:
This will raise a TypeError: `exceptions must derive from BaseException`

Reply:
Changed to a module.fail_json message (seems like better practice anyways)